service/src/java/org/apache/hive/service/cli/CLIService.java

```diff
@@ -457,9 +469,34 @@ public OperationStatus getOperationStatus(OperationHandle opHandle)
     }
     OperationStatus opStatus = operation.getStatus();
     LOG.debug(opHandle + ": getOperationStatus()");
+    opStatus.setJobProgressUpdate(progressUpdateLog(getProgressUpdate, operation));
     return opStatus;
   }
 
+  private JobProgressUpdate progressUpdateLog(boolean isProgressLogRequested, Operation operation) {
+    if (isProgressLogRequested && canProvideProgressLog()) {
+        if (OperationType.EXECUTE_STATEMENT.equals(operation.getType())) {
+          SessionState sessionState = operation.getParentSession().getSessionState();
+          try {
+            while (sessionState.getProgressMonitor() == null && !operation.isFinished()) {
+              Thread.sleep(100L); // sleep for 100 ms
```

Comment:
this is a very lightweight check, it would be OK to poll faster on this. Maybe sleep every 10 MS instead, so we don't add another 100MS of latency ?